lib/ruby_lsp/requests/code_lens.rb

```diff
@@ -0,0 +1,61 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    # ![Code lens demo](../../misc/code_lens.gif)
+    #
+    # This feature is currently experimental. Clients will need to pass `experimentalFeaturesEnabled`
+    # in the initialization options to enable it.
+    #
+    # The
+    # [code lens](https://microsoft.github.io/language-server-protocol/specification#textDocument_codeLens)
+    # request informs the editor of runnable commands such as tests
+    #
+    # # Example
+    #
+    # ```ruby
+    # # Run
+    # class Test < Minitest::Test
+    # end
+    # ```
+
+    class CodeLens < BaseRequest
+      sig do
+        params(
+          document: Document,
+        ).void
+      end
+      def initialize(document)
+        super(document)
+        @results = T.let([], T::Array[Interface::CodeLens])
+        @path = T.let(document.uri.delete_prefix("file://"), String)
+      end
+
+      sig { override.returns(T.all(T::Array[Interface::CodeLens], Object)) }
+      def run
+        visit(@document.tree) if @document.parsed?
+        @results
+      end
+
+      sig { override.params(node: SyntaxTree::ClassDeclaration).void }
+      def visit_class(node)
+        class_name = node.constant.constant.value
+        if class_name.end_with?("Test")
+          add_code_lens(node, [@path, class_name, "ruby -Itest " + @path])
```

Comment:
Should this be prefixed with `bundle exec`?

Reply:
As far as I know, it shouldn't be necessary. This is how we run things with `debug` as well. @st0012 any thoughts?

Reply:
_Most_ of the time it won't matter, but there's a chance that someone could have a newer, incompatible version of a gem compared to what's in `Gemfile.lock`.

Reply:
Our current workflow with VS Code creates a custom bundle even for projects that don't have a `Gemfile`. I agree with Andy, we can do something like this
```ruby
class Codelens < BaseRequest
  BASE_COMMAND = T.let(File.exist?("Gemfile.lock") ? "bundle exec ruby" : "ruby", String)
end
```